app/src/main/java/com/codesoom/assignment/application/AuthenticationService.java

```diff
@@ -0,0 +1,76 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.domain.User;
+import com.codesoom.assignment.domain.UserRepository;
+import com.codesoom.assignment.dto.UserLoginData;
+import com.codesoom.assignment.errors.EmailNotFoundException;
+import com.codesoom.assignment.errors.UnauthorizedException;
+import com.codesoom.assignment.errors.WrongPasswordException;
+import com.codesoom.assignment.utils.JwtUtil;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.security.SignatureException;
+import org.springframework.stereotype.Service;
+
+/**
+ * 유저 인증 로직 담당.
+ */
+@Service
+public class AuthenticationService {
+
+    private final JwtUtil jwtUtil;
+    private final UserRepository userRepository;
+
+    public AuthenticationService(JwtUtil jwtUtil, UserRepository userRepository) {
+        this.jwtUtil = jwtUtil;
+        this.userRepository = userRepository;
+    }
+
+    /**
+     * 토큰을 decode한 후 속성정보를 반환한다.
+     *
+     * @param accessToken 인증토큰
+     * @return 속성 정보
+     * @throws UnauthorizedException 인증에 실패한 경우
+     */
+    public Long parsetoken(String accessToken) {
+        if (accessToken == null || accessToken.isBlank()) {
+            throw new UnauthorizedException(accessToken);
+        }
+        try {
+            Claims claims = jwtUtil.decode(accessToken);
+            return claims.get("userId", Long.class);
+        } catch (SignatureException e) {
+            throw new UnauthorizedException(accessToken);
+        }
+    }
+
+    /**
+     * 로그인 데이터로 유저 정보 조회후, userID로 인증토큰을 반환한다.
+     *
+     * @param loginData 로그인 데이터
+     * @return 인증토큰
+     * @throws WrongPasswordException 비밀번호가 틀린 경우
+     */
+    public String createToken(UserLoginData loginData) {
+        User user = findUserByEmail(loginData);
+        if (!user.authenticate(loginData.getPassword())) {
+            throw new WrongPasswordException(loginData);
+        }
+        return jwtUtil.encode(user.getId());
+    }
+
+    /**
+     * 주어진 로그인 데이터와 일치하는 유저를 반환한다.
+     *
+     * @param loginData 로그인 데이터
+     * @return 유저
+     * @throws EmailNotFoundException 이메일이 저장되어있지 않은 경우
```

Comment:
"이메일이 저장되어 있지 않은 경우"와 "이메일을 찾지 못한 경우"는 다릅니다. 컴퓨터를 다룰 때에는 무엇이 "없다"라고 단언하면 곤란할 수 있습니다. 이게 조회를 하는 순간에 DB에 무슨 일이 생겨서 조회가 실패한 건지, 정말로 데이터가 존재하지 않아서 실패한 건지, 아니면 그냥 내 컴퓨터가 인터넷이 끊어져서 실패한 건지 아주 짧은 시간 동안에 확실히 알아내기란 매우 어렵습니다.

그래서 수많은 에러 메시지가 "~을 찾지 못했습니다."라고 작성됩니다.

```suggestion
     * @throws EmailNotFoundException 로그인 정보의 이메일을 찾지 못한 경우
```

Reply:
미처 그런경우는 생각지 못했네요. 감사합니다~!